src/openms/source/MATH/STATISTICS/ROCCurve.cpp

```diff
@@ -87,17 +98,57 @@ namespace OpenMS
         return 0.5;
       }
 
-      score_clas_pairs_.sort(simsortdec());
+      sort();
+      double prevscore = -std::numeric_limits<double>::infinity();
+      UInt prevTP = 0;
+      UInt prevFP = 0;
+      UInt truePos = 0;
+      UInt falsePos = 0;
+      double area = 0.0;
+      for (auto const& pair : score_clas_pairs_)
+      {
+        //since it is sorted we do not need abs here.
+        //TODO not sure if this makes much difference to an equality
+        if ((pair.first - prevscore) > 1e-8)
+        {
+          area += trapezoidal_area(falsePos,prevFP,truePos,prevTP);
+          prevscore = pair.first;
+          prevTP = truePos;
+          prevFP = falsePos;
+        }
+        pair.second ? ++truePos : ++falsePos;
+      }
+      area += trapezoidal_area(falsePos,prevFP,truePos,prevTP);
+      // scale to unit square
+      area /= truePos * falsePos;
+
+      // update internals
+      pos_ = truePos;
+      neg_ = falsePos;
+
+      return area;
+    }
+
+    double ROCCurve::rocN(Size N)
+    {
+      if (score_clas_pairs_.size() < N)
+      {
+        std::cerr << "ROCCurve::rocN() : unsuitable dataset (not enough false positives)\n";
+        return -1;
```

Comment:
is this return value documented in header?

Reply:
yes